src/RepoAuditor/EntryPoint.py

```diff
@@ -34,18 +48,182 @@ def list_commands(self, *args, **kwargs):  # pylint: disable=unused-argument
-@app.command("Version")
-def Version() -> None:
-    """Prints the version of the package."""
+def _HelpEpilog() -> str:
+    content: list[str] = []
+
+    for module in _all_modules:
+        arguments: list[str] = []
+
+        for arg_name, type_info in module.GetDynamicArgDefinitions().items():
+            if isinstance(type_info, TyperEx.TypeDefinitionItem):
+                python_type = type_info.python_type
+                parameter_info = type_info.parameter_info
+            elif isinstance(type_info, tuple):
+                python_type = type_info[0]
+                parameter_info = type_info[1]
+            else:
+                python_type = type_info
+                parameter_info = None
+
+            arguments.append(
+                "    {arg_name:<32} {type_description:<8} {help}".format(
+                    arg_name=f"--{module.name}{ARGUMENT_SEPARATOR}{arg_name}",
+                    type_description=python_type.__name__,
+                    help="" if parameter_info is None else parameter_info.help,
+                ),
+            )
+
+        final_arguments = TextwrapEx.Indent(
+            "\n".join(arguments),
+            4,
+            skip_first_line=True,
+        )
+
+        content.append(
+            textwrap.dedent(
+                f"""\
+                {module.name}
+                {"-" * len(module.name)}
+                {module.description}
+
+                Command Line Arguments:
+                    {final_arguments}
+                """,
+            ),
+        )
+
+    return (
+        textwrap.dedent(
+            """\
+        Module Information
+        ==================
+
+        {}
+        """,
+        )
+        .format(
+            "\n".join(content),
+        )
+        .replace("\n", "\n\n")
+    )
+
+
+# ----------------------------------------------------------------------
+def _VersionCallback(value: bool) -> None:
+    if value:
+        sys.stdout.write(f"RepoAuditor v{__version__}\n")
+        raise typer.Exit()
+
+
+# ----------------------------------------------------------------------
+@app.command(
+    "EntryPoint",
+    context_settings={"allow_extra_args": True, "ignore_unknown_options": True},
+    help=__doc__,
+    epilog=_HelpEpilog(),
+    no_args_is_help=False,
+)
+def EntryPoint(
+    ctx: typer.Context,
+    version: Annotated[  # pylint: disable=unused-argument
+        bool,
+        typer.Option(
+            "--version",
+            help="Display the version of this tool and exit.",
+            callback=_VersionCallback,
+            is_eager=True,
+        ),
+    ] = False,
+    excludes: Annotated[
+        list[str],
+        typer.Option(
+            "--exclude",
+            help=f"Module or requirement names to exclude from execution; line command line arguments, requirement names must include the module name as a prefix (e.g. 'ModuleName{ARGUMENT_SEPARATOR}RequirementName'). This value can be provided multiple times.",

```

Comment:
is that meant to be "**like** command line arguments..."?

Reply:
Good catch - thank you! Fix incoming....